src/ast/dml.rs

```diff
@@ -32,14 +32,42 @@ use sqlparser_derive::{Visit, VisitMut};
 pub use super::ddl::{ColumnDef, TableConstraint};
 
 use super::{
-    display_comma_separated, display_separated, query::InputFormatClause, Assignment, ClusteredBy,
-    CommentDef, Expr, FileFormat, FromTable, HiveDistributionStyle, HiveFormat, HiveIOFormat,
-    HiveRowFormat, Ident, InsertAliases, MysqlInsertPriority, ObjectName, OnCommit, OnInsert,
-    OneOrManyWithParens, OrderByExpr, Query, RowAccessPolicy, SelectItem, Setting, SqlOption,
-    SqliteOnConflict, StorageSerializationPolicy, TableEngine, TableObject, TableWithJoins, Tag,
-    WrappedCollection,
+    display_comma_separated, display_separated, operator_classes::IndexOperatorClass,
+    query::InputFormatClause, Assignment, ClusteredBy, CommentDef, Expr, FileFormat, FromTable,
+    HiveDistributionStyle, HiveFormat, HiveIOFormat, HiveRowFormat, Ident, IndexType,
+    InsertAliases, MysqlInsertPriority, ObjectName, OnCommit, OnInsert, OneOrManyWithParens,
+    OrderByExpr, Query, RowAccessPolicy, SelectItem, Setting, SqlOption, SqliteOnConflict,
+    StorageSerializationPolicy, TableEngine, TableObject, TableWithJoins, Tag, WrappedCollection,
 };
 
+/// Index column type.
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub struct IndexColumn {
+    pub column: OrderByExpr,
+    pub operator_class: Option<IndexOperatorClass>,
+}
+
+impl From<OrderByExpr> for IndexColumn {
+    fn from(column: OrderByExpr) -> Self {
+        Self {
+            column,
+            operator_class: None,
+        }
+    }
+}
```

Comment:
hmm I'm thinking having an `OrderByExpr` -> `IndexColumn` could be arbitrary and potentially cause issues/mismatches if either struct evolves going forward, it might be better to manually construct `IndexColumn` where this was used?

Reply:
Removed.